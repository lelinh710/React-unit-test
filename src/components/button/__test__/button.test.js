import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../button';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-render';

afterEach(cleanup);

it('render without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div);
})

// it('render button correctly', () => {
//     const {getByTestID} =  render(<Button label='Click me!'/>);
//     expect(getByTestID('button')).toHaveTextContent('Click me!');
// })

// it('render button correctly', () => {
//     const {getByTestID} =  render(<Button label='save'/>);
//     expect(getByTestID('button')).toHaveTextContent('save');
// })

it('match snapshot', () => {
    const tree = renderer.createRenderer(<Button label='save'/>).toJSON();
    expect(tree).toMatchSnapshot();
})

