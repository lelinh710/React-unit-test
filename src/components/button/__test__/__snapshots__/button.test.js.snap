

exports[`match snapshot`] = `
        <div data-testid='button' className='button-style' >
            save
        </div>
        `;